src/test/regress/expected/shard_rebalancer.out

```diff
@@ -104,6 +133,37 @@ SELECT create_distributed_table('dist_table_test_2', 'a');
 
 (1 row)
 
+-- replicate_table_shards should fail when the hostname GUC is set to a non-reachable node
+ALTER SYSTEM SET citus.local_hostname TO 'foobar';
+SELECT pg_reload_conf();
+ pg_reload_conf
+---------------------------------------------------------------------
+ t
+(1 row)
+
+SELECT pg_sleep(1); -- wait to make sure the config has changed before running the GUC
+ pg_sleep
+---------------------------------------------------------------------
+
+(1 row)
+
+SET citus.shard_replication_factor TO 2;
+SELECT replicate_table_shards('dist_table_test_2',  max_shard_copies := 4,  shard_transfer_mode:='block_writes');
+NOTICE:  Copying shard xxxxx from localhost:xxxxx to localhost:xxxxx ...
+ERROR:  connection to the remote node foobar:57636 failed with the following error: could not translate host name "foobar" to address: <system specific error>
```

Comment:
Long ago I naively wrote a test that assumed 1.1.1.1 was always going to be unreachable, but it was pretty interesting to see when that test started failing. Now I'm curious when someone will register the foobar TLD and give it an A record :).

Reply:
I can make it `foobar.local`. I believe the ICANN has put the `local` tld on a reserved list for its use in local networks.